migration-roadmap/README.md

```diff
@@ -0,0 +1,174 @@
+
+- **Clear starting point**: Entry point for development work that identifies specific areas needing adjustment
+- **Gap analysis**: Differences that would prevent Spectrum 2 CSS from working in SWC
+- **Reference materials**: Links to CSS and React implementations, and previous migration work
+
+## What we analyze
+
+### Component specifications
+
+We extract and document:
+
+- **CSS selectors**: All selectors from `metadata.json` in the spectrum-css `spectrum-two` branch
+- **Passthroughs**: CSS passthrough properties
+- **Modifiers**: CSS modifier classes
+- **SWC attributes**: Properties with `@property` decorators in TypeScript
+- **SWC slots**: Slot patterns from render methods
+
+### DOM structure comparison
+
+We analyze markup from three sources:
+
+1. **Spectrum Web Components**: Current HTML structure from web component `render()` method
+2. **Legacy CSS**: HTML structure from spectrum-css `main` branch template files
+3. **Spectrum 2 CSS**: HTML structure from spectrum-css `spectrum-two` branch template files
+
+This three-way comparison reveals:
+
+- DOM changes specific to Spectrum 2 features (Legacy → Spectrum 2)
+- Structural differences between CSS and SWC implementations
+- Additional work needed for SWC render methods beyond new features
+
+### CSS to SWC mapping
+
+We create mapping tables that connect:
+
+- **CSS selectors** to **SWC attributes or slots**
+- **Variants** (CSS selectors with `--` notation) typically map to component attributes
+- **Base elements** (selectors with single `-`) typically map to slots
+- **Status tracking** for implementation gaps
+
+### Implementation gaps
+
+The analysis surfaces:
+
+- **Missing from SWC**: CSS features without web component equivalents
+- **Missing from CSS**: Web component features without CSS support
+- **New for Spectrum 2**: Features that need to be added for 2nd generation
+- **Deprecated**: Features being removed in Spectrum 2
+
+## Documentation structure
+
+Each component follows this format:
+
+```markdown
+# [Component name] migration roadmap
+
+## Component specifications
+
+### CSS
+
+- CSS selectors (collapsible)
+- Passthroughs (collapsible)
+- Modifiers (collapsible)
+
+### SWC
+
+- Attributes (collapsible)
+- Slots (collapsible)
+
+## Comparison
+
+### DOM structure changes
+
+- Spectrum Web Components (collapsible)
+- Legacy CSS main branch (collapsible)
+- Spectrum 2 CSS spectrum-two branch (collapsible)
+- Diff: Legacy → Spectrum 2, based on changes made to template in CSS migration work (collapsible, if changes exist)
+
+### CSS => SWC mapping
+
+| CSS selector | Attribute or slot | Status                                                  |
+| ------------ | ----------------- | ------------------------------------------------------- |
+| ...          | ...               | Implemented/Missing from WC/Missing from CSS/Deprecated |
+
+## Summary of changes
+
+### CSS => SWC implementation gaps
+
+### CSS Spectrum 2 changes
```

Comment:
Should we add a note somewhere that some of this information might be summarized in the component's changelog in the `spectrum-two` branch?